src/fc_checks.c

```diff
@@ -232,3 +232,181 @@ struct check_result *check_file_context_types_exist(__attribute__((unused)) cons
+	 * to that, ifndef is not mutually exclusive both in theory and practice so that   *
+	 * assumption is made here as well. Those are the condition under which this       *
+	 * function will return true:                                                      *
+	 *                                                                                 *
+	 * 1)two duplicates whereby none of them is within any ifdef/ifndef.               *
+	 *                                                                                 *
+	 * 2)two duplicates whereby one of them is within an ifdef/ifndef and the other    *
+	 *   one is not within any.                                                        *
+	 *                                                                                 *
+	 * 3)two two duplicates whereby both are within the same ifdef/ifndef. by both     *
+	 *   here I'm not referring to literally be defined under the same conditional     *
+	 *   although it would also return true in this case, rather I'm referring to the  *
+	 *   parameters or conditions being the same. so if you had the same specification *
+	 *   under distro_redhat in two different files,it would treat them as being under *
+	 *   the same conditional and return true.                                         *
+	 *                                                                                 *
+	 * 4)two duplicates whereby one is within an ifdef, the other within an ifndef     *
+	 *   and the conditions differ i.e ifdef(`distro_gentoo',` and                     *
+	 *   ifndef(`distro_redhat',`                                                      *
+	 *                                                                                 *
+	 * 5)two duplicates whereby both are within ifndef and the conditions differ i.e   *
+	 *   ifndef(`distro_gentoo',` and ifndef(`distro_redhat',`                         *
+	 *                                                                                 *
+	 * For any other conditions, it returns false                                      *
+	 *                                                                                 *
+	 * For Future improvement we can consider evaluating ifdef/ifndef instead.         *
+	 * the conditional_data which is a field from fc_entry contains a boolean field    *
+	 * called 'state' specifically to handle that for future improvement. Currently    *
+	 * this field is not used it's just a place holder.                                *
+	 ***********************************************************************************/
+
+	if (!entry_one->conditional || !entry_two->conditional) {
+		return true;
+	} else if (entry_one->conditional && entry_two->conditional) {
+		if (entry_one->conditional->flavor == entry_two->conditional->flavor
+				&& (!strcmp(entry_one->conditional->condition,
+						entry_two->conditional->condition)
```

Comment:
Please align multiple arguments to a function call together.  You can tab up to the previous indentation and then use spaces for alignment